core/modules/czar/Czar.cc

```diff
@@ -201,19 +200,23 @@ Czar::submitQuery(std::string const& query,
 
         result.resultTable = resultTableName;
         result.messageTable = asyncLockName;
-
+        if (not resultTableName.empty()) {
+            result.resultQuery = std::string("SELECT * FROM ") + resultTableName;
+        }
     } else {
-
+        result.messageTable = lockName;
         if (not uq->getResultTableName().empty()) {
             result.resultTable = resultDb + "." + uq->getResultTableName();
+            result.resultQuery = std::string("SELECT * FROM ") + result.resultTable;
+            auto&& orderBy = uq->getProxyOrderBy();
+            if (not orderBy.empty()) {
+                result.resultQuery += std::string(" ") + orderBy;
```

Comment:
For performance-obsessed maniacs like my cat :cat2: this may look better:

    result.resultQuery += ' ';
    result.resultQuery += orderBy;